drivers/pwm_dac/pwm_dac.c

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2020 Christian Amsüss <chrysn@fsfe.org>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_pwm_dac
+ * @{
+ *
+ * @file
+ * @brief       DAC implementation based on a PWM device
+ *
+ * @author      Christian Amsüss <chrysn@fsfe.org>
+ * @}
+ */
+
+#include "pwm_dac.h"
+#include <periph/dac.h>
+#include <periph/pwm.h>
+#include <assert.h>
+#include <bitarithm.h>
+
+#ifdef HAVE_DAC_T
+#error "pwm_dac is not usable on platforms with a DAC, especially such that" \
+    " define their own dac_t."
+#endif
+
+#if UINT_MAX != 0xffffffff
+#error "pwm_dac requires a 32-bit default dac_t to fit all internal details" \
+    " into the dac_t value."
+#endif
```

Comment:
Right now `periph_dac` is only implemented on 32 bit platforms anyway, so you wouldn't change anything by doing

```patch
--- a/drivers/include/periph/dac.h
+++ b/drivers/include/periph/dac.h
@@ -64,7 +64,7 @@ extern "C" {
  * @brief   Define default DAC type identifier
  */
 #ifndef HAVE_DAC_T
-typedef unsigned int dac_t;
+typedef uint32_t dac_t;
 #endif
 
 /**
```

and dropping this check.

8-bit MCUs can then still define `HAVE_DAC_T` if they want a smaller DAC type.